src/mono/mono/metadata/image.c

```diff
@@ -2029,6 +2070,11 @@ mono_image_open_full (const char *fname, MonoImageOpenStatus *status, gboolean r
 	return mono_image_open_a_lot (alc, fname, status, refonly, FALSE);
 }
 
+/**
+ * mono_image_open_a_lot_parameterized
+ * this API is not culture aware, so if we load a satellite assembly for one culture by name
+ * via this API, and then try to load it with another culture we will return the first one.
+ */
 static MonoImage *
 mono_image_open_a_lot_parameterized (MonoLoadedImages *li, MonoAssemblyLoadContext *alc, const char *fname, MonoImageOpenStatus *status, gboolean refonly, gboolean load_from_context, gboolean *problematic)
```

Comment:
I don't understand this comment.  `mono_image_open_a_lot_parametrized` doesn't know anything about cultures because it's a low-level API that is only concerned with file paths, not assembly names.  You should never have the same path name for two different satellite assemblies.